### PR TITLE
fix: Ensure user id for early track events

### PR DESCRIPTION
**Pull request by mutdmour**
## Summary
- fix bug where early track events don't have user id
- ensure identify is set when user is fetched before those track events to set user id

## Related Linear tickets, Github issues, and Community forum posts
Fixes ADO-2438

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/plugins/telemetry/index.ts [hunk 1]
```diff
@@ -146,6 +144,10 @@ export class Telemetry {
 		}
 	}
 
+	reset() {
```
Comment (MiloradFilipovic) [inline packages/editor-ui/src/plugins/telemetry/index.ts:147]: There is also a call to `rudderStack.reset` in `identify` action ([here](https://github.com/n8n-io/n8n/blob/8fd044eae03fa314a7ab35a16967dab09499c888/packages/editor-ui/src/plugins/telemetry/index.ts#L99)). Do we still need that?

**Review by MiloradFilipovic — APPROVED** [review]
Looks good. I left a comment with a question but might be unrelevant

File: packages/editor-ui/src/plugins/telemetry/index.ts (hunk 1 above)
Comment (mutdmour) [reply packages/editor-ui/src/plugins/telemetry/index.ts:147]: we still need that when initializing RS if user is not logged in.

**Comment by janober** [comment]
Got released with [`n8n@1.61.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.61.0)

**Pull request merged**